encord/client.py

```diff
@@ -504,10 +506,35 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
-
-        return AddPrivateDataResponse.from_dict(response)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        logger.info(f"Dataset creation job started with upload_job_id={process_hash}.")
+        logger.info("SDK process can be terminated, this will not affect successful dataset creation.")
+
+        while True:
+            polling_response = None
+
+            for _ in range(3):
+                try:
+                    polling_response = self._querier.basic_getter(
+                        DatasetDataLongPolling,
+                        self._config.resource_id,
+                        payload={"process_hash": process_hash},
+                    )
+                except requests.exceptions.RequestException as e:
+                    logger.error(e)
```

Comment:
It is not an `error` here. It would be a `warning` at most. Also just logging `e` is weird. First of all, this is what `logger.error` already does for you, second, this is meant to be for users. They will not know what is happening without any context. You'd need to add a more informative message.

Reply:
done